src/core/annotation.js

```diff
@@ -485,7 +485,7 @@ var LinkAnnotation = (function LinkAnnotationClosure() {
     data.annotationType = AnnotationType.LINK;
 
     var action = dict.get('A');
-    if (action) {
+    if (isDict(action)) {
```

Comment:
thats the first check in isDict. so what you actually only avoid is the method call to that util function. there are many more instances were we prefer the util method cal over the trivial implementation inside. 
like
```
function isArray(v) {
  return v instanceof Array;
}
```